cdap-client/src/main/java/co/cask/cdap/client/ProgramClient.java

```diff
@@ -75,10 +76,12 @@ public ProgramClient(ClientConfig config, RESTClient restClient, ApplicationClie
     this.applicationClient = applicationClient;
   }
 
+  public ProgramClient(ClientConfig config, RESTClient restClient) {
+    this(config, restClient, new ApplicationClient(config, restClient));
+  }
+
   public ProgramClient(ClientConfig config) {
```

Comment:
nit: can we make this the first constructor, the one on line 79 the second, and the first one here the third. Just makes it easier to read.